### DISCUSSION TITLE
Build error for Erigon Windows

### DISCUSSION THREAD

**Discussion by josephversace** [erigontech/erigon#2980] [Q&A] [answered]
I am running into an error trying to build windows binaries. The powershell script fails here.

# crawshaw.io/sqlite
blocking_step.c: In function 'unlock_notify_cb':
blocking_step.c:42:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for(int i=0; i < nArg; i++) {
  ^
I have tried modifying the following line in wmake.ps1 but that did not help.

    -D CMAKE_C_COMPILER:PATH=""$(Join-Path $chocolateyBinPath "gcc.exe -std=c99")"" `

Anyone able to assist?

**Comment by AndreaLanfranchi** [comment]
Please open a PowerShell prompt and report :

- the output of  `gcc.exe --version`
- the output of  `choco list --local-only`
- the output of `($env:path).split(";")`
- check if you have any other version of C/C++ compilers installed

Also report **what branch** are you trying to build: `stable` or `devel` ?

**Comment by josephversace** [comment]
Thank you Andrea,

PS C:\erigon>  gcc.exe --version
gcc.exe (GCC) 4.8.3

PS C:\erigon> choco list --local-only
choco : The term 'choco' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ choco list --local-only
+ ~~~~~
    + CategoryInfo          : ObjectNotFound: (choco:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException


PS C:\erigon> ($env:path).split(";")
C:\WINDOWS\system32
C:\WINDOWS
C:\WINDOWS\System32\Wbem
C:\WINDOWS\System32\WindowsPowerShell\v1.0\
C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\
C:\Program Files\Microsoft SQL Server\110\Tools\Binn\
c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\
c:\Program Files\Microsoft SQL Server\110\DTS\Binn\
c:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\ManagementStudio\
c:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\
C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\
C:\Program Files\Microsoft SQL Server\120\Tools\Binn\
C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\
C:\Program Files\Microsoft SQL Server\130\Tools\Binn\
C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\
C:\Program Files\VisualSVN Server\bin
C:\Program Files\TortoiseSVN\bin
C:\Program Files\dotnet\
C:\Program Files\nodejs\
C:\Program Files\Microsoft\Web Platform Installer\
C:\Program Files\Docker
C:\Program Files\Git\cmd
C:\Program Files\Go\bin
C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps
C:\Users\Administrator\.dotnet\tools
C:\Users\Administrator\AppData\Roaming\npm
C:\Users\Administrator\AppData\Local\Programs\Microsoft VS Code\bin
C:\MinGW\bin

I was trying to build the devel branch I think. I wanted to use it with ethtx so I need the mainnet_archive with debug mode

**Comment by josephversace** [comment]
I have installed Chocolately now and the updated output is below

PS C:\erigon> choco list --local-only
Chocolatey v0.11.3
chocolatey 0.11.3

Still the same error though.

**Reply by AndreaLanfranchi** [reply]
> I have installed Chocolately now and the updated output is below
> 
> PS C:\erigon> choco list --local-only Chocolatey v0.11.3 chocolatey 0.11.3
> 
> Still the same error though.

When you install chocolatey you also need to run
`choco install cmake make mingw`

And above all remove/uninstall any other version MinGW installed on computer or, at the very least, remove those from `%PATH%` env variable.

**Comment by AndreaLanfranchi** [comment, marked as answer]
Ok I see. Problem is your gcc version is too old.
We need at least gcc 10 (sorry we'll amend readme)

I suggest to 
- Uninstall your version of gcc (I see you installed mingw) and remove `C:\MinGW\bin` from `%PATH%` environment variable
- Install chocolatey package manager following the instructions of bullet 3 in the [Windows section](https://github.com/ledgerwatch/erigon#windows)
- eventually retry the build

**Comment by josephversace** [comment]
Thank you. That solved it. Appreciate it,
